flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaConnectorOptions.java

```diff
@@ -118,8 +118,8 @@ public class KafkaConnectorOptions {
                     .asList()
                     .noDefaultValue()
                     .withDescription(
-                            "Topic names from which the table is read. Either 'topic' or 'topic-pattern' must be set for source. "
-                                    + "Option 'topic' is required for sink.");
+                            "Topic name(s) to read data from when the table is used as source. It also supports topic list for source by separating topic by semicolon like 'topic-1;topic-2'. Note, only one of 'topic-pattern' and 'topic' can be specified for sources. "
+                                    + "When the table is used as sink, the topic name is the topic to write data. It also supports topic list for sinks. The provided topic-list is treated as a allow list of valid values for the `topic` metadata column, if not provided, any value is valid. If the 'topic' option isn't specified, or a list is provided, for sink table, 'topic' metadata column is writable and must be specified.");
```

Comment:
> if not provided, any value is valid

how about supporting `topic-pattern` in sink as well? This would align with source and `*` wildcard pattern stands for "any value is valid".

Reply:
After discussion, we thought the `topic-pattern` would be a more intuitive semantic than `topic=null` for supporting 'any value' for topic metadata column.